plugins/hls-class-plugin/src/Ide/Plugin/Class/Types.hs

```diff
@@ -81,30 +115,101 @@ data BindInfo = BindInfo
       -- ^ SrcSpan of the binding name
     }
 
-rules :: Recorder (WithPriority Log) -> Rules ()
-rules recorder = do
-    define (cmapWithPrio LogShake recorder) $ \GetInstanceBindTypeSigs nfp -> do
-        tmr <- use TypeCheck nfp
-        hsc <- use GhcSession nfp
-        result <- liftIO $ instanceBindType (hscEnv <$> hsc) (tmrTypechecked <$> tmr)
-        pure ([], result)
+getInstanceBindLensRule :: Recorder (WithPriority Log) -> Rules ()
+getInstanceBindLensRule recorder = do
+    defineNoDiagnostics (cmapWithPrio LogShake recorder) $ \GetInstanceBindLens nfp -> runMaybeT $ do
+        tmr@(tmrRenamed ->  (hs_tyclds -> tycls, _, _, _)) <- useMT TypeCheck nfp
+        (InstanceBindTypeSigsResult allBinds) <- useMT GetInstanceBindTypeSigs nfp
```

Comment:
This is the only place this rule is called... so is there any reason not to just inline the logic here???

Reply:
It is used also for code actions, hence the need to create a seperate rule for lens specific stuff.